File: algorithms/[1486]数组异或操作/solution.ts
// topics = ["位运算"]

/**
 * Bit Manipulation
 * time O(n), space O(1)
 */
function xorOperation(n: number, start: number): number {
  let res = 0;
  for (let i = 0; i < n; i += 1) {
    res ^= start + 2 * i;
  }

  return res;
}

File: algorithms/[1486]数组异或操作/solution.c
// topics = ["位运算"]

/**
 * 模拟 time O(n), space O(1)
 */
int xorOperation(int n, int start)
{
    int res = 0;
    for (int i = 0; i < n; i++)
    {
        res ^= start + 2 * i;
    }

    return res;
}